### PR TITLE
[Compatibility] Added changes for backward compatibility

### PR DESCRIPTION
**Pull request by kunal642**
This PR will fix the issues related to old version and new version compatibility.
**Issues fixed:**
1. Schema file name was different in one of the previous versions.
2. bucket number was not supported in the previous versions.
3. table parameters were stored in lower case while in the current version we are reading in camel case.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 No
 - [X] Any backward compatibility impacted?
 No
 - [X] Document update required?
No
 - [X] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1255/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2479/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2499/

**Comment by jackylk** [comment]
Please add description for this PR

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1275/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2664/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2512/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1295/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2680/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala [hunk 1]
```diff
@@ -291,4 +291,52 @@ object CarbonScalaUtil {
...
+    val keyParts = parameters.getOrElse("spark.sql.sources.options.keys.numparts", "0").toInt
+    if (keyParts == 0) {
+      parameters
+    } else {
+      var keyStr = ""
+      for(i <- 0 until keyParts) {
+        keyStr += parameters(s"spark.sql.sources.options.keys.part.$i")
+      }
+      val finalProperties = scala.collection.mutable.Map.empty[String, String]
+      keyStr.split(",") foreach {
```
Comment (manishgupta88) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala]: Please check the logic here.  Keystr is not appended explicitly with , but splitting is done based on comma

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala [hunk 2]
```diff
@@ -291,4 +291,52 @@ object CarbonScalaUtil {
...
+
+  /**
+   * If the table is from an old store then the table parameters are in lowercase. In the current
+   * code we are reading the parameters as camel case.
+   * This method will convert all the schema parts to camel case
+   *
+   * @param parameters
+   * @return
+   */
+  def getDeserializedParameters(parameters: Map[String, String]): Map[String, String] = {
```
Comment (manishgupta88) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala]: there is no calling point for this method

File: core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java [hunk 3]
```diff
@@ -231,7 +231,22 @@ public String getSchemaFilePath() {
    * @return schema file path
    */
   public static String getSchemaFilePath(String tablePath) {
-    return tablePath + File.separator + METADATA_DIR + File.separator + SCHEMA_FILE;
+    return getActualSchemaFilePath(tablePath);
+  }
+
+  private static String getActualSchemaFilePath(String tablePath) {
+    String metaPath = tablePath + File.separator + METADATA_DIR;
```
Comment (manishgupta88) [inline core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java]: use CarbonCommonConstants.FILE_SEPARATOR instead of File.separator

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3221/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1987/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3181/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**